src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -2228,7 +2228,12 @@ public async Task<(long, object)> PutConnectionString(TransactionOperationContex
                     break;
 
                 case ConnectionStringType.Sql:
-                    command = new PutSqlConnectionStringCommand(JsonDeserializationCluster.SqlConnectionString(connectionString), databaseName, raftRequestId);
+                    // RavenDB-21784 - Replace obsolete MySql provider name
+                    var deserializedSqlConnectionString = JsonDeserializationCluster.SqlConnectionString(connectionString);
+                    if (deserializedSqlConnectionString.FactoryName == "MySql.Data.MySqlClient")
+                        deserializedSqlConnectionString.FactoryName = "MySqlConnector.MySqlConnectorFactory";
```

Comment:
Let's create a notification in the studio saying that old connector is deprecated and we defaulted to the current one:

https://issues.hibernatingrhinos.com/issue/RavenDB-21716/Unhandled-SQL-error-can-crash-RavenDB#focus=Comments-67-1046779.0-0

Reply:
done https://github.com/ravendb/ravendb/pull/17822/files#diff-343189e415881abff6c312fa21edc9c5b752335a99df35ee03bf75419bb5fd60R2236-R2238